tests/extendrtests/src/rust/build.rs

```diff
@@ -1,12 +1,56 @@
-use std::env;
+use std::path::PathBuf;
 
 fn main() {
-    // TODO: I couldn't find any nice way to add the condition based on the R version
-    // except for using libR-sys just for "DEP_R_*" envvars.
-    let major = env::var("DEP_R_R_VERSION_MAJOR").unwrap();
-    let minor = env::var("DEP_R_R_VERSION_MINOR").unwrap();
+    println!("cargo:rerun-if-env-changed=R_HOME");
+    println!("cargo::rerun-if-changed=build.rs");
 
-    if &*major >= "4" && &*minor >= "3" {
+    let r_home_env = std::env::var_os("R_HOME");
+    let r_home_env = r_home_env.map(PathBuf::from);
+
+    // prefer `R_HOME` if set, otherwise use the `PATH` available R
+    let mut r_version = if let Some(r_home_env) = r_home_env {
+        std::process::Command::new(r_home_env.join("R"))
+    } else {
+        std::process::Command::new("R")
+    };
+    let r_version = r_version
+        .arg("CMD")
+        .args(["config", "--version"])
+        .output()
+        .expect("failed to run `R CMD config --version`");
+    assert!(r_version.status.success());
+    use std::io::BufRead as _;
+    let r_version_line = r_version
+        .stdout
+        .lines()
+        .next()
+        .expect("there were no outputs from `R CMD config`")
+        .expect("failed to read the line with R version");
+    let raw_r_version = r_version_line
+        .split(':')
+        .nth(1)
+        .unwrap()
+        .trim()
+        // ignore commit
+        // only capture the first the major.minor.patch part of the version
+        .split_ascii_whitespace()
+        .next()
+        .unwrap();
+    dbg!(raw_r_version);
```

Comment:
Do we need debug statement in 'production' code?

Reply:
I recently found out, that calls to `dbg!` gets piped to a file `tests/extendrtests/src/rust/target/debug/build/extendrtests-44614d0cd19b0ae0/stderr` (hashing is part of the incremental compilation I think), which means you'd have a file that contains (this doesn't get printed to user, terminal or elsewhere)
```
[build.rs:39:5] raw_r_version = "4.5.0"
```

So that's helpful.

I've removed a ton of these `dbg!` calls and one slipped out, but I think it is nice with some reporting, somewhere neat (doesn't pollute terminal).